pkg/config/datafileprojectconfig/config.go

```diff
@@ -173,7 +185,13 @@ func (c DatafileProjectConfig) GetGroupByID(groupID string) (entities.Group, err
 func NewDatafileProjectConfig(jsonDatafile []byte) (*DatafileProjectConfig, error) {
 	datafile, err := Parse(jsonDatafile)
 	if err != nil {
-		logger.Error("Error parsing datafile.", err)
+		logger.Error("Error parsing datafile", err)
+		return nil, err
+	}
+
+	if !inSlice(datafileVersions, datafile.Version) {
+		err = errors.New("unsupported datafile version")
+		logger.Error(fmt.Sprintf("Version %s of datafile not supported", datafile.Version), err)
```

Comment:
Will be useful to say unable to initialize/update or something along those lines.